client/asset/interface.go

```diff
@@ -72,7 +72,7 @@ type Wallet interface {
 	// Fund selects coins for use in an order. The coins will be locked, and will
 	// not be returned in subsequent calls to Fund or calculated in calls to
 	// Available, unless they are unlocked with ReturnCoins.
-	FundOrder(uint64, *dex.Asset) (Coins, error)
+	FundOrder(value, lots uint64, assetInfo *dex.Asset) (Coins, error)
```

Comment:
Since this can be confusing ("why do I have to specify value and lots?"), the API could frame this as, say, `maxSwaps` instead of `lots` since what this really gets to is maximum number of swap contract txns to account for in funding the fees.

I suspect that we neglected the full relevance of the fields of the `dex.Asset` (`coin.LotSize` doesn't apply to buy orders) because we didn't spell out just what they were for and which fields were used and how.  So as an alternative to passing `dex.Asset`, we could rethink the `FundOrder` api to possibly use a `OrderFundSpec` or `SwapAsset` struct with just the relevant fields named explicitly for their purpose in funding an order.  With this, `value` could be a separate arg or part of this hypothetical struct.   But if we document `FundOrder` in detail about the purpose of the passed `dexAsset`, which fields are pertinent to order funding, and any other args, then I have no problem with continuing to use `dex.Asset`.